python/cugraph_alg.py

```diff
@@ -0,0 +1,56 @@
+import mgp
+import pandas
+import sys
+
+try:
+    import cugraph
+except ImportError as import_error:
+    sys.stderr.write(
+        "RAPIDS cuGraph is not installed, please build Memgraph from source with Python Conda interpreter where cuGraph is installd ."
+    )
+    raise import_error
+
+
+class GraphDefinition:
+    SOURCE = "source"
+    TARGET = "target"
+
+    VERTEX = "vertex"
+    PAGERANK = "pagerank"
+
+
+def _get_mg_data(mg_graph: mgp.Graph) -> pandas.DataFrame:
+    graph_data = {GraphDefinition.SOURCE: [], GraphDefinition.TARGET: []}
+    for from_vertex in mg_graph.vertices:
+        for edge in from_vertex.out_edges:
+            to_vertex = edge.to_vertex
+            graph_data[GraphDefinition.SOURCE].append(from_vertex.id)
+            graph_data[GraphDefinition.TARGET].append(to_vertex.id)
+
+    df = pandas.DataFrame(graph_data)
+    return df
+
+
+@mgp.read_proc
+def pagerank(
+    context: mgp.ProcCtx,
+) -> mgp.Record(vertex_id=mgp.Number, pagerank=mgp.Number):
```

Comment:
I think it's smarter to put `vertex=mgp.Any` instead of `vertex_id=mgp.Number`, by having that, both IDs and whole node objects will be supported (somehow I got nodes out of this call, haven't gone into details why yet).

Reply:
This seems to be a mistake, the return value should be `mgp.Vertex` according to the written code below